tests/common/fixtures/advanced_reboot.py

```diff
@@ -115,9 +115,14 @@ def __buildTestbedData(self):
 
         self.rebootData['dut_hostname'] = self.mgFacts['minigraph_mgmt_interface']['addr']
         self.rebootData['dut_mac'] = hostFacts['ansible_Ethernet0']['macaddress']
-        self.rebootData['dut_username'] = hostFacts['ansible_env']['SUDO_USER']
         self.rebootData['vlan_ip_range'] = self.mgFacts['minigraph_vlan_interfaces'][0]['subnet']
         self.rebootData['dut_vlan_ip'] = self.mgFacts['minigraph_vlan_interfaces'][0]['addr']
+
+        invetory = self.duthost.host.options['inventory'].split('/')[-1]
+        secrets = self.duthost.host.options['variable_manager']._hostvars[self.duthost.hostname]['secret_group_vars']
```

Comment:
I faced `KeyError: ('secret_group_vars',) ` when tried to run `platform/test_advanced_reboot.py::test_fast_reboot` which uses this fixture.

In the next two lines, you tried to get `sonicadmin_user` and `sonicadmin_password`  from `secrets`
I have `sonicadmin_user` and `sonicadmin_password` directly in `self.duthost.host.options['variable_manager']._hostvars[self.duthost.hostname]`

The values are defined in  `ansible/group_vars/lab/secrets.yml`.

I, Maybe, missed something in testbed definition, but do we really need to get  `sonicadmin_user` and `sonicadmin_password` from `secret_group_vars`? Maybe there is a mistake? 

If No, then where secret_group_vars should be defined?

@tahmed-dev  Could you please help me?

Reply:
@msosyak Thanks for your question.

Those secrets are defined in secrets.json file (ansible/group_vars/all/secrets.json). Here is the format of the file:

`{
    .
    .
    "secret_group_vars": {
        "str": {
            "ansible_ssh_pass": "",
            "ansible_become_pass": "",
            "sonicadmin_user": "",
            "sonicadmin_password": "",
            "sonicadmin_password": "",
            "sonicadmin_initial_password": "",
            "lab_admin_pass": "",
            "ptf_host_user": "",
            "ptf_host_pass": "",
            "fanout_admin_user": "",
            "fanout_admin_password": ""
        },
    },
    .
    .
}
`
Where str in the above file is the name of the inventory file. You may wanna change it the name of the inventory file used in your setup.